### PR TITLE
[WB-5474]: Custom Charts don't work most of the time on colab and jupyter in visual studio code

### PR DESCRIPTION
**Pull request by KyleGoyette**
https://wandb.atlassian.net/browse/WB-5474

Description
-----------

Modifies how the session_history is saved in jupyter notebooks.

Testing
-------

Locally

### CODE REVIEW

File: wandb/jupyter.py [hunk 1]
```diff
@@ -359,8 +359,9 @@ def save_history(self):
                 },
             )
             state_path = os.path.join("code", "_session_history.ipynb")
-            wandb.run.config["_wandb"]["session_history"] = state_path
-            wandb.run.config.persist()
+            wandb.run._config_callback(
+                val=state_path, key=("_wandb", "session_history")
+            )
```
Comment (raubitsj) [inline wandb/jupyter.py:364]: nice catch, can you instead put a private method on wandb.run (or config)
I think we used to have one called `run._set_config_wandb` 

That will be cleaner than referring to the callback here.

Comment (KyleGoyette) [reply wandb/jupyter.py:364]: Can do!

File: wandb/jupyter.py [hunk 2]
```diff
@@ -359,7 +359,7 @@ def save_history(self):
                 },
             )
             state_path = os.path.join("code", "_session_history.ipynb")
-            wandb.run._config_callback(
+            wandb.run._set_config_wandb(
                 val=state_path, key=("_wandb", "session_history")
```
Comment (raubitsj) [inline wandb/jupyter.py:363]: I think the common case is just setting a single key inside _wandb so the interface should probably be:
key: str, val: Any

hide the nested key stuff from the caller.

```
wandb.run._set_config_wandb("session_history", state_path)
```

And you have a mypy lint error

Comment (KyleGoyette) [reply wandb/jupyter.py:363]: Ah yeah, you're right.

**Review by raubitsj — APPROVED** [review]
LGTM

**Pull request merged**